# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/nsDNSService2.h
# Commit: 6f4ca967aa44
# Full Hash: 6f4ca967aa446cf24f4224576f51803c982f3d7b
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2024-06-14 20:49:45
# Regressor Bug: 1901990
# File Overlap Count: 2
# Description:
#   Bug 1901990 - Introduce a pref for setting a mock HTTPS RR, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213559
# ==============================================================================

diff -r 8a2bdb44382a -r 6f4ca967aa44 netwerk/dns/nsDNSService2.h
--- a/netwerk/dns/nsDNSService2.h	Fri Jun 14 13:27:36 2024 +0000
+++ b/netwerk/dns/nsDNSService2.h	Fri Jun 14 13:42:37 2024 +0000
@@ -17,6 +17,7 @@
 #include "nsString.h"
 #include "nsTHashSet.h"
 #include "nsHashKeys.h"
+#include "mozilla/Atomics.h"
 #include "mozilla/Mutex.h"
 #include "mozilla/Attributes.h"
 #include "TRRService.h"
@@ -109,8 +110,8 @@
   RefPtr<nsHostResolver> mResolver;
   nsCOMPtr<nsIIDNService> mIDN;
 
-  // mLock protects access to mResolver, mLocalDomains, mIPv4OnlyDomains and
-  // mFailedSVCDomainNames
+  // mLock protects access to mResolver, mLocalDomains, mIPv4OnlyDomains,
+  // mFailedSVCDomainNames, and mMockHTTPSRRDomain.
   mozilla::Mutex mLock MOZ_UNANNOTATED{"nsDNSServer.mLock"};
 
   // mIPv4OnlyDomains is a comma-separated list of domains for which only
@@ -118,6 +119,8 @@
   // a per-domain basis and work around broken DNS servers. See bug 68796.
   nsCString mIPv4OnlyDomains;
   nsCString mForceResolve;
+  nsCString mMockHTTPSRRDomain;
+  mozilla::Atomic<bool, mozilla::Relaxed> mHasMockHTTPSRRDomainSet{false};
   bool mBlockDotOnion = false;
   bool mNotifyResolution = false;
   bool mOfflineLocalhost = false;